qcelemental/models/molecule.py

```diff
@@ -401,18 +407,7 @@ def compare(self, other):
         else:
             raise TypeError("Comparison molecule not understood of type '{}'.".format(type(other)))
 
-        match = True
-        match &= np.array_equal(self.symbols, other.symbols)
-        match &= np.allclose(self.masses, other.masses, atol=(10**-MASS_NOISE))
-        match &= np.array_equal(self.real, other.real)
-        match &= np.array_equal(self.fragments, other.fragments)
-        match &= np.allclose(self.fragment_charges, other.fragment_charges, atol=(10**-CHARGE_NOISE))
-        match &= np.array_equal(self.fragment_multiplicities, other.fragment_multiplicities)
-
-        match &= np.allclose(self.molecular_charge, other.molecular_charge, atol=(10**-CHARGE_NOISE))
-        match &= np.array_equal(self.molecular_multiplicity, other.molecular_multiplicity)
-        match &= np.allclose(self.geometry, other.geometry, atol=(10**-GEOMETRY_NOISE))
-        return match
+        return self.get_hash() == other.get_hash()
```

Comment:
Do we want to allow `compare` to stay in to allow tunable tolerances? I can think of a few use cases, but then again we have `compare_recursive` which can handle this.

@loriab any thoughts here?

Reply:
I can think of lots o'ways ppl want to judge mol equality (connectivity-wise, geometry-wise, geometry-w/i-tol-wise, geometry-ignoring-frame-and-orientation-wise, equal-except-isotope, equal-except-chgmult, equal-ignoring-fragmentation, equal-ignoring-ghostiness, equal-except-different-bases-on-atoms, equal-but-for-labels, equal-but-for-atom-order). Looks like existing `compare` is focusing on just variations in numerics. I don't think I use `models.Molecule.compare`, so no harm to my downstream in dropping it.

Someday, may want to coalesce some of the above into a tunable fn (with some presets like "is_isotope"), but I'm fine with dropping for now to focus on the broadly used preset hash. Being a built-in, that'll not be tunable, so a separate fn will make sense someday.